include/wasmtime.hh

```diff
@@ -1742,30 +1810,61 @@ class Val {
 
   wasmtime_val_t val;
 
-  Val() : val({.kind = WASMTIME_I32, .of = {.i32 = 0}}) {}
+  Val() : val{} {
+    val.kind = WASMTIME_I32;
+    decltype(val.of) valunion{};
```

Comment:
How come this separate variable is necessary? Could this do `val.of.i32 = 0` directly instead? (similar for below constructors too)

Reply:
Yeah, you are right ... by some reason much more complex way )))
Updated